bindata/network/ovn-kubernetes/ovnkube-node.yaml

```diff
@@ -2,7 +2,11 @@
 kind: DaemonSet
 apiVersion: apps/v1
 metadata:
+  {{ if eq .OVN_DPU_MODE "dpu-host" }}
```

Comment:
What happens if this switch is flipped? Do we support enabling this in running clusters? Is every node expected to have a DPU?

Reply:
Yes, current idea is to enable this in running cluster, not changing CNO API.
We'd start by deploying the full ovn-k8s CNI on the dpu host, then configure the dpu host to run in the dpu-host mode.

If the switch is flipped,  full-mode (default) ovnkube-node will be re-deployed to the dpu host,  which may conflict with the ovnkube-node running on dpu.

Reply:
I see. How much do you care if the change is disruptive?

So, one problem is that renaming a daemonset is the same as a delete and add, which is unsynchronized. This means you could have two ovnkube-nodes running on the same host. It would be better if you could do this without renaming the daemonset.

Is there some way for ovn-kube node to autodetect that it should be in dpu mode? Maybe by reading the labels on the Node object itself? Then you don't have to do any work when you roll out the changes.

Reply:
> So, one problem is that renaming a daemonset

We're not renaming it; see the corresponding code change. When DPU offload is enabled, CNO will render this particular template file twice, once with `OVN_DPU_MODE=full` to generate the `ovnkube-node` DaemonSet for the "dumb" nodes, and once with `OVN_DPU_MODE=dpu-host` to generate the `ovnkube-node-dpu-host` DaemonSet for the "smart" nodes.

> This means you could have two ovnkube-nodes running on the same host.

This part is still a problem though.